src/main/flight/pid.c

```diff
@@ -323,7 +319,32 @@ void pidUpdateTpaFactor(float throttle)
         tpaRate = pidRuntime.tpaLowMultiplier * (pidRuntime.tpaLowBreakpoint - tpaArgument);
     }
 
-    float tpaFactor = 1.0f - tpaRate;
+    return 1.0f - tpaRate;
+}
+
+void pidUpdateTpaFactor(float throttle, const pidProfile_t *pidProfile)
+{
+    // don't permit throttle > 1 & throttle < 0 ? is this needed ? can throttle be > 1 or < 0 at this point
+    throttle = constrainf(throttle, 0.0f, 1.0f);
+    float tpaFactor;
+
+#ifdef USE_WING
+    const float tpaArgument = isFixedWing() ?  getWingTpaArgument(throttle) : throttle;
+
+    switch (pidProfile->tpa_curve_type) {
+        case TPA_CURVE_HYPERBOLIC:
+            tpaFactor = pwlInterpolate(&pidRuntime.tpaCurvePwl, tpaArgument);
+            break;
+        case TPA_CURVE_CLASSIC:
+        default:
+            tpaFactor = getTpaFactorClassic(tpaArgument);    
+    }
+#else // USE_WING is undefined
```

Comment:
TPA_CURVE_HYPERBOLIC is quite generic - maybe it's useful without wing too (with simple throttle as argument)

Reply:
made it behind `USE_ADVANCED_TPA`